src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ConditionController.kt

```diff
@@ -0,0 +1,427 @@
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "The licence for this ID was not found.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun updateBespokeConditions(
+    @PathVariable("licenceId") licenceId: Long,
+    @Valid @RequestBody
+    request: BespokeConditionRequest,
+  ) {
+    licenceConditionService.updateBespokeConditions(licenceId, request)
+  }
+
+  @Tag(name = Tags.LICENCE_CONDITIONS)
+  @PostMapping(value = ["/id/{licenceId}/additional-condition/{conditionType}"])
+  @PreAuthorize("hasAnyRole('SYSTEM_USER', 'CVL_ADMIN')")
+  @Operation(
+    summary = "Add additional condition to the licence.",
+    description = "Add additional condition to the licence. " +
+      "This does not include accompanying data per condition. Existing conditions which appear on " +
+      "the licence will be unaffected. More than one condition with the same code can be added " +
+      "Requires ROLE_SYSTEM_USER or ROLE_CVL_ADMIN.",
+    security = [SecurityRequirement(name = "ROLE_SYSTEM_USER"), SecurityRequirement(name = "ROLE_CVL_ADMIN")],
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Set of additional conditions added",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad request, request body must be valid",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised, requires a valid Oauth2 token",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "The licence for this ID was not found.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun addAdditionalCondition(
+    @PathVariable licenceId: Long,
+    @PathVariable conditionType: String,
```

Comment:
Not something to resolve now as I appreciate it's a lift-and-shift, but do we actually need this? Or is it just for url mapping?

Reply:
I think ideally we'd change the url but its a bit faffy